modules/core/src/smithy4s/RefinementProvider.scala

```diff
@@ -82,6 +83,12 @@ object RefinementProvider {
   implicit def mapLengthConstraint[K, V]: Simple[Length, Map[K, V]] =
     new LengthConstraint[Map[K, V]](_.size)
 
+  implicit def enumLengthConstraint[E <: Enumeration.Value]: Simple[Length, E] =
```

Comment:
**question:** should we also have `@pattern`?

Reply:
`@idRef` would be nasty. If someone ever creates an enum that matches the ShapeId format, I'll personally go to their house and remove their github account.

Reply:
> question: should we also have @pattern? 

Is it allowed ? you mentioned range and length, but ...

Reply:
I didn't check the others! 😅